From 32232af1c1422f01adf0404926167ad715cb32c8 Mon Sep 17 00:00:00 2001
From: Harris Hancock <harris@cloudflare.com>
Date: Fri, 28 Jun 2024 14:19:25 +0100
Subject: [PATCH 11/22] Revert TracedReference deref API removal

This previously-deprecated API was removed in 7cd2b0c434bd5ea7bd253a97958c7348557fd72e, but we don't know how to safely replace our usage of it yet.
---
 include/v8-traced-handle.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/include/v8-traced-handle.h b/include/v8-traced-handle.h
index 3eb8e7835d7..b6b958d426b 100644
--- a/include/v8-traced-handle.h
+++ b/include/v8-traced-handle.h
@@ -134,6 +134,17 @@ class BasicTracedReference : public TracedReferenceBase {
         const_cast<BasicTracedReference<T>&>(*this));
   }
 
+  V8_DEPRECATED("Use Get to convert to Local instead")
+  V8_INLINE T* operator->() const {
+#ifdef V8_ENABLE_CHECKS
+    CheckValue();
+#endif  // V8_ENABLE_CHECKS
+    return this->template value<T>();
+  }
+
+  V8_DEPRECATED("Use Get to convert to Local instead")
+  V8_INLINE T* operator*() const { return this->operator->(); }
+
  private:
   /**
    * An empty BasicTracedReference without storage cell.
-- 
2.43.0

